redfish-core/lib/pcie.hpp

```diff
@@ -67,8 +67,36 @@ static inline void
         {
             for (const auto& [serviceName, interfaceList] : serviceMap)
             {
-                std::string devName =
-                    sdbusplus::message::object_path(pcieDevicePath).filename();
+                sdbusplus::message::object_path path(pcieDevicePath);
+                sdbusplus::message::object_path parentPath = path.parent_path();
+                sdbusplus::message::object_path parentofParentPath =
+                    parentPath.parent_path();
+                std::string devName;
+
+                /**
+                 * @brief Workaround to handle duplicate PCI device list
+                 *
+                 * retrieve PCI device endpoint information and if path is
+                 * ~/chassisN/io_moduleN/slotN/adapterN then, replace redfish
+                 * PCI device as "chassisN_io_moduleN_slotN_adapterN"
+                 *
+                 */
+
+                if (parentofParentPath.parent_path().filename().find(
+                        "chassis") != std::string::npos)
+                {
+                    devName = parentofParentPath.parent_path().filename() + "_";
+                }
+                if (parentofParentPath.filename().find("io_module") !=
```

Comment:
might we replace some of this 'parentofParentPath' things with dbus::utility::getNthStringFromPath?
[Example here](https://github.com/ibm-openbmc/bmcweb/blob/e2f2ca0519409c45b7e28564fa3187952eadb8b5/redfish-core/lib/managers.hpp#L757)